apps/settings/js/panels/full_developer_mode/panel.js

```diff
@@ -0,0 +1,56 @@
+define(function(require) {
+  'use strict';
+
+  var SettingsService = require('modules/settings_service');
+  var SettingsCache = require('modules/settings_cache');
+  var DialogService = require('modules/dialog_service');
+  var Panel = require('modules/panel');
+
+  /**
+   * The panel can be invoked by triggering an activity:
+   * new MozActivity({
+   *   name: 'configure',
+   *   data: {
+   *     target: 'device',
+   *     section: 'full-developer-mode'
+   *   }
+   * })
+   */
+  return function ctor_full_dev_panel() {
+    return Panel({
+      onInit: function(panel) {
+        this._panel = panel;
+        SettingsCache.getSettings((results) => {
+          var devMenuEnabled = !!results['developer.menu.enabled'];
+          if (devMenuEnabled) {
+            this._showFullDevModeWarning();
+          } else {
+            this._showEnableDevMenuWarning();
+          }
+        });
+      },
+      _close: function() {
+        SettingsService.back();
+      },
+      _showFullDevModeWarning: function() {
+        var warningDialog = this._panel.querySelector('.full-dev-mode-warning');
+        var okBtn = warningDialog.querySelector('button[type="submit"]');
+        var cancelBtn = warningDialog.querySelector('button[type="reset"]');
+
+        warningDialog.hidden = false;
+        cancelBtn.onclick = this._close.bind(this);
```

Comment:
I think we don't have to rebind scope to `this` here because in `_close` method, we don't use anything about `this`.